strawberryfields/engine.py

```diff
@@ -476,9 +479,9 @@ def run(self, program, *, args=None, compile_options=None, **kwargs):
         if not isinstance(program, collections.abc.Sequence) and program.type == "tdm":
             # At this time we do not support lists of tdm programs
             result._all_samples = reshape_samples(
-                result.all_samples, program.measured_modes, program.N
+                result.all_samples, program.measured_modes, program.N, program.timebins
             )
-            result._samples = np.array(list(result.all_samples.values()))
+            result._samples = np.array(list(result.all_samples.values())).transpose(1, 0, 2)
```

Comment:
How come this transposition is needed? Can it simply be absorbed into the `reshape_samples` function?

Reply:
It's because of the shape change we decided to have for both `all_samples`, `{spatial mode: (shots, timebins)}`, and `samples`, `(shots, spatial modes, timebins)`, along with that `reshape_samples` only actually reshapes and returns `all_samples`. `samples` is created here, outside of the `reshape_samples` function, although I guess this could be move inside of the `reshape_samples` instead, if we want, so that it returns the tuple `(all_samples, samples)` instead.

Reply:
Worth leaving a comment on this.